reporter/src/main/kotlin/utils/ZeroBasedIntSequenceGenerator.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.reporter.utils
+
+import com.fasterxml.jackson.annotation.ObjectIdGenerator
+import com.fasterxml.jackson.annotation.ObjectIdGenerators
+
+/**
+ * This [ObjectIdGenerator] works similar to [ObjectIdGenerators.IntSequenceGenerator] with the difference that the
+ * initialValue is 0 instead of 1.
+ */
+class ZeroBasedIntSequenceGenerator @JvmOverloads constructor(
+    private val scope: Class<*> = Any::class.java,
+    @field:Transient private var nextValue: Int = -1
+) : ObjectIdGenerator<Int>() {
+    companion object {
+        private const val INITIAL_VALUE = 0
```

Comment:
How about just using `0` as default value for `nextValue` and get rid of this companion object?
If not applicable consider moving the constant to file scope.

Reply:
I wanted to keep this implementation as close as possible to Jackson's `ObjectIdGenerators.IntSequenceGenerator`, which also initializes `nextValue` with `-1` in the zero argument constructor. Though I have to admit I do not know why. Right now the only difference between the two classes is the initial value.

Reply:
Thanks, I'm ok with that.